### PR TITLE
Add metric warning support (announcement bar and DM)

### PR DESCRIPTION
**Pull request by catalintomai**
#### Summary
Add sysadmin warning support (announcement bar and DM) for multiple metrics (here - if the number of active users metric is above a certain limit).

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-25042

#### Related PRs
webapp: https://github.com/mattermost/mattermost-webapp/pull/5447
server: https://github.com/mattermost/mattermost-server/pull/14483

### CODE REVIEW

File: src/reducers/requests/admin.ts [hunk 1]
```diff
@@ -408,6 +408,16 @@ function disablePlugin(state: RequestStatusType = initialRequestState(), action:
     );
 }
 
+function sendWarnMetricAck(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
```
Comment (jespino) [inline src/reducers/requests/admin.ts:411]: We are no longer tracking the request unless is strictly needed, for example use the request completion to trigger something in other side of the code. But if the code making the call is the one that is reacting to the changes we only need to use the async/await (or promise) result for showing that.

Comment (jespino) [reply src/reducers/requests/admin.ts:411]: Removing this is making some changes here and there, not only this reducer.

File: src/actions/admin.test.js [hunk 2]
```diff
@@ -1209,4 +1209,15 @@ describe('Actions.Admin', () => {
 
         const state = store.getState();
     });
+
+    it('sendWarnMetricAck', async () => {
+        const warnMetricAck = {
+            metric1: true,
+        };
+        nock(Client4.getBaseRoute()).
+            get('/warn_metrics/ack').
```
Comment (jespino) [inline src/actions/admin.test.js:1218]: Shouldn't it be mocking the `post` call instead of the `get` call?

File: src/actions/general.ts [hunk 3]
```diff
@@ -185,6 +185,24 @@ export function getRedirectLocation(url: string): ActionFunc {
...
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let data;
+        try {
+            data = await Client4.getWarnMetricsStatus();
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            return {error};
+        }
+
+        dispatch(batchActions([
```
Comment (jespino) [inline src/actions/general.ts:198]: this `batchActions` is not needed

File: src/actions/general.test.js [hunk 4]
```diff
@@ -132,6 +132,24 @@ describe('Actions.General', () => {
...
+            metric2: false,
+        };
+
+        nock(Client4.getBaseRoute()).
+            get('/warn_metrics/status').
+            query(true).
+            reply(200, responseData);
+
+        await Actions.getWarnMetricsStatus()(store.dispatch, store.getState);
+        await TestHelper.wait(100);
```
Comment (jespino) [inline src/actions/general.test.js:147]: I'm not sure this `wait` is needed here, for sure shouldn't be needed, but I'm not sure if we need more changes anywhere to make it work properly.

File: src/actions/admin.test.js (hunk 2 above)
Comment (catalintomai) [reply src/actions/admin.test.js:1218]: yes it should, will update

File: src/actions/general.test.js (hunk 4 above)
Comment (catalintomai) [reply src/actions/general.test.js:147]: I'll remove it, but right now it looks to be used in some places and not in others, so I erred on the caution side.

File: src/actions/general.ts (hunk 3 above)
Comment (catalintomai) [reply src/actions/general.ts:198]: sure

File: src/reducers/requests/admin.ts (hunk 1 above)
Comment (catalintomai) [reply src/reducers/requests/admin.ts]: removed the request tracking defines+related.

**Comment by srkgupta** [comment]
Hi @catalintomai 
I am beginning to test this now. Can you please resolve all the merge conflicts here. Thanks.

**Comment by srkgupta** [comment]
This PR is awaiting some of the issues address in the dependent PR:
https://github.com/mattermost/mattermost-server/pull/14483

**Pull request merged**